### DISCUSSION TITLE
How to "shutting down gracefully"

### DISCUSSION THREAD

**Discussion by WangHan2014** [bluenviron/mediamtx#2841] [Questions]
### Question

i'm building my own library on Android,and i want to know how to manual shutdown it but "os.exit",it'll kill whole application.

Thanks！

**Comment by aler9** [comment]
Please reopen the issue and provide details that allow to replicate the problem, in particular how you are embedding the server into your application. Otherwise it's impossible to understand what are you trying to achieve.
